### PR TITLE
shell: Never connect to remote machines automatically

### PR DESCRIPTION
**Pull request by mvollmer**
Release note:
----
### Shell: SSH connections to remote machines are only opened when necessary

Cockpit remembers all remote hosts of a session and used to immediately connect to all of them when starting the next session.  Now a SSH connection is only opened when navigating to the remote machine for the first time.

### CODE REVIEW

**Comment by mvollmer** [comment]
The initial state of a remote machine looks like this in this draft:

image

Let's improve that by not making it look like an error.

**Comment by mvollmer** [comment]
> Let's improve that by not making it look like an error.

This is no longer necessary.  Now we connect when navigating, so people will not see this curtain.

File: pkg/shell/base_index.js [hunk 1]
```diff
@@ -790,10 +790,10 @@ function Index() {
 
     self.ready = function () {
         $(window).on("popstate", function(ev) {
-            self.navigate(ev.state, true);
+            self.navigate(ev.state);
```
Comment (mvollmer) [inline pkg/shell/base_index.js:793]: This change is not necessary, I'd say.

**Comment by mvollmer** [comment]
GREEEEEN!

File: test/verify/check-host-switching [hunk 2]
```diff
@@ -389,6 +395,23 @@ class TestHostSwitching(MachineCase, HostSwitcherHelpers):
...
+        # And and connect to a second machine
+        b.click("#hosts-sel button")
+        self.add_machine(b, "10.111.113.2")
+        b.click("a[href='/@10.111.113.2']")
+        b.wait_visible("iframe.container-frame[name='cockpit1:10.111.113.2/system']")
+        self.assertIn("admin", m2.execute("loginctl"))
+        b.click("#hosts-sel button")
+        b.click("a[href='/@localhost']")
+        b.relogin()
+        time.sleep(60)
```
Comment (martinpitt) [inline test/verify/check-host-switching:413]: That's certainly a long time, but I see that we want to make really sure. :grin: But the test is destructive, so it doesn't  hold up the nd ones.

**Review by martinpitt — APPROVED** [review]
Thank you, amazing work! This was certainly not easy to figure out, but it's surprisingly clear to review. I had expected more noise.

**Pull request merged**